pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -99,9 +98,19 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 			attrs.enableSriov = true
 			requireChange = true
 		}
-		if fwCurrent.linkType != EthLinkType && fwCurrent.linkType != PreconfiguredLinkType && fwCurrent.linkType != UknownLinkType {
-			attrs.linkType = EthLinkType
+		if iface.LinkType != EthLinkType && iface.LinkType != InfinibandLinkType {
+			err = fmt.Errorf("mellanox-plugin OnNodeStateChange(): Not supported link type: %s", iface.LinkType)
+			return false, false, err
+		}
+		if fwCurrent.linkType == PreconfiguredLinkType || fwCurrent.linkType == UknownLinkType {
+			err = fmt.Errorf("mellanox-plugin OnNodeStateChange(): Preconfigured NIC or Unkown link type: %s", fwCurrent.linkType)
```

Comment:
What's the meaning of `Preconfigured NIC`? The error message looks a bit confusing, why a `Preconfigured NIC` would lead  to an error?

Reply:
In some of mlnx's cards the `LINK_TYPE_P*` is reconfigured to IB or ETH and can't be changed

Reply:
I would separate the two cases

in case of unknown link type:
`err = fmt.Errorf("mellanox-plugin OnNodeStateChange(): Unknown Link type : %s", fwCurrent.linkType)`

in case of a PreConfigredLinkType:
`err = fmt.Errorf("mellanox-plugin OnNodeStateChange(): Network card does not support Link type change.")`